engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -469,12 +475,37 @@ object ProgramExecutionSupport {
                 .getOrElse(expressionValue.getClass)
             )
           )
-          ctx.executionService.callFunctionWithInstrument(
-            visualization.cache,
-            visualization.module,
-            visualization.callback,
-            expressionValue +: visualization.arguments: _*
-          )
+          visualization.config.expression match {
+            case Api.VisualizationExpression.Text(_, expression) =>
+              val compiler = ctx.executionService.getContext.getCompiler
+              val ensoRootNode =
+                expressionNode.getRootNode.asInstanceOf[EnsoRootNode]
+              val compilerConfig = CompilerConfig(
+                autoParallelismEnabled = false,
+                warningsEnabled        = false,
+                isStrictErrors         = true
+              )
+              val moduleContext = ModuleContext(
+                ensoRootNode.getModuleScope.getModule,
+                compilerConfig
+              )
+              val inlineContext = InlineContext(
+                moduleContext,
+                compilerConfig,
+                Some(ensoRootNode.getLocalScope),
+                Some(false)
+              )
+              val callback = compiler.runInline(expression, inlineContext).get
+              val fn       = Function.thunk(callback.getRootNode.getCallTarget, null)
```

Comment:
According to @kustosz we are supposed to construct the function just like the [eval node does](https://github.com/enso-org/enso/blob/08c9ecb54037396a6ea26247c37b2bccb5ac1b4f/engine/runtime/src/main/java/org/enso/interpreter/node/expression/debug/EvalNode.java#L120C61-L120C61) in its `uncached` case.

That means we have to have `callerInfo.getFrame()`